### ISSUE TITLE
Talk 12: Jason Hemann

### ISSUE THREAD

**Issue by rrnewton** [wonks/ICFP_rehearsal_feedback#12]

**Comment by cgswords** [comment]
Love the 10-foot font

**Comment by samth** [comment]
Use of Racket is implied by the `()`, you don't have to have it on the slides. :)

**Comment by samth** [comment]
Syntax highlighting would be nice.

**Comment by cgswords** [comment]
- We can probably get rid of slide 2...
- Love the fake repl
- Maybe call valof 'eval' for those schemers?

**Comment by cgswords** [comment]
6 minutes in, we're finishing up background

**Comment by samth** [comment]
More visuals, less text.

**Comment by laurmcarter** [comment]
Re syntax highlighting, color distinction between bound variables and quoted data would be nice.

**Comment by cgswords** [comment]
That bulleted list of the type inferencer, program generator, CL reducer, could make sweet visuals

**Comment by laurmcarter** [comment]
... or just distinguishing quoted data, for that matter.

**Comment by ccshan** [comment]
I'm confused about the organization of the introduction. You talk about miniKanren, then microKanren, then back to miniKaren on the next (book cover) slide -- why move back and forth, and more importantly, what is the goal of your talk?

**Comment by cgswords** [comment]
I feel like this entire constraint discussion could have happened before introducing microkanren.

**Comment by ccshan** [comment]
For syntax highlighting, you could follow the typesetting of the Reasoned Schemer book?

**Comment by cgswords** [comment]
Yes, you go through why miniKanren hurts for constraints, and then you could lead into microkanren and talk about adding constraints to it!

**Comment by ccshan** [comment]
Is the headline "tiny implementation of logic programming now features tiny implementation of constraints"? (Because tiny is teachable) Then put the desiderata slide as FIRST slide.

**Comment by cgswords** [comment]
:fireworks: ^^^^^^ what ken said

**Comment by samth** [comment]
Agree 100% with @ccshan and @cgswords.

**Comment by samth** [comment]
Never use a fixed-width font for text.

**Comment by cgswords** [comment]
Okay we're getting into the good things, and we're already at 10 minutes (the Constraint Recipe slide). That intro should tighten up.

**Comment by samth** [comment]
How did the font get so small?

**Comment by ccshan** [comment]
Maybe one way to demonstrate the pedagogical success of microKanren is to jump right into explaining it (without going through miniKanren - just say that it is too big).

**Comment by cgswords** [comment]
@ccshan "I'm not even going to bother explaining miniKanren; it's just too big. I'm going to show you some microKanren, and you'll have no problem understanding it" ?

**Comment by cgswords** [comment]
The syntax-rule could get a nice picture to explain it.

**Comment by cgswords** [comment]
The same for your call/fresh. A little logic puzzle diagram could be cute? Or some sort of reasoning as we move through the constraints there.

**Comment by ccshan** [comment]
The claim of "extensible" makes me want a list of extensions that have been made. I assume: booleans (subsuming SAT solver), numbers (subsuming some kind of SMT), etc.

**Comment by samth** [comment]
Complicated macros on your slide seem unnecessary. Also, why does `S0` need to be unhygenic?

**Comment by ccshan** [comment]
Put "120 lines" claim on the first slide too.

**Comment by cgswords** [comment]
Using the System, example 1, hits 20 minutes

**Comment by cgswords** [comment]
Write a pretty-printer, use it, sell it as a front-end?

**Comment by samth** [comment]
I feel like `match` could be used usefully somehow in the type inference example.

**Comment by cgswords** [comment]
@samth are you proposing generating match expanders for mK constraints? Because that's a neat idea.

**Comment by ccshan** [comment]
Time: 22:22
